backend/metadata_writer/src/metadata_helpers.py

```diff
@@ -276,6 +277,7 @@ def create_new_execution_in_existing_run_context(
             EXECUTION_RUN_ID_PROPERTY_NAME: metadata_store_pb2.Value(string_value=run_id),
             EXECUTION_COMPONENT_ID_PROPERTY_NAME: metadata_store_pb2.Value(string_value=instance_id), # should set to task ID, not component ID
         },
+        custom_properties=custom_properties,
```

Comment:
Should this be properties or custom_properties?

Reply:
it was a custom property in tfx

Reply:
Cool. Then this lgtm

Reply:
> Should this be properties or custom_properties?

Hmm. You're right. It's a property. i thought it was custom_property based on https://github.com/kubeflow/pipelines/issues/3462#issue-595705018 .

I thought that TFX was already moving all non-user properties to custom_properties. But I guess they only did that for Artifact and did not for Executions yet.
I think they wanted to upgrade Executions too eventually.

@Bobgy Would you be OK with checking the custom property as well or would you prefer for me to log this as a declared typed property? The property is a bit more problematic due to schema changes that can cause conflicts. Setting custom_properties does not require schema and does not cause cconflicts.

Reply:
> it was a custom property in tfx

I'm not 100% sure about that: https://github.com/tensorflow/tfx/blob/ac8a421117c0eeed38bab2fd9a4fb7c18b68376f/tfx/orchestration/kubeflow/kubeflow_metadata_adapter_test.py#L105

Reply:
Ohhh, sorry I think I remembered it wrong

Reply:
https://github.com/Bobgy/pipelines/blob/39b0a22a4e1f6a93ad2f3128ecc6bd26ee8e94d2/frontend/src/lib/OutputArtifactLoader.ts#L475
UI is also reading it from property

Reply:
> UI is also reading it from property

Can we add ` || execution.getCustomPropertiesMap().get('kfp_pod_name')` in the UX? This will make Frontend more future-proof and compatible with the future TFX changes.

Reply:
Sure, actually new code is already reading from both places: https://github.com/kubeflow/pipelines/pull/3457/files#diff-e5adac73b344b2e45e343675eb04838bR260

The helper method getResourceProperty tries to read from both property and custom property

Reply:
> The helper method getResourceProperty tries to read from both property and custom property

I'm not sure about that. It seems to me that it reads from only one of those places (based on an argument).
https://github.com/kubeflow/frontend/blob/2a39c3fe405e033ea0b262a40d9dc2e7f9a8153a/src/mlmd/Utils.tsx#L34

Reply:
Yes, you are right. I was confused with a different method: https://github.com/kubeflow/frontend/blob/2a39c3fe405e033ea0b262a40d9dc2e7f9a8153a/src/mlmd/Utils.tsx#L42.

If we duplicate a property in both custom/non-custom like in https://github.com/kubeflow/frontend/blob/2a39c3fe405e033ea0b262a40d9dc2e7f9a8153a/src/mlmd/Api.ts#L9, we can get it in both places.

Reply:
I was very confused what's the difference between properties and custom properties. Looking at current data tfx pipelines log. Most of executions' properties are properties, while most of artifacts' properties are custom properties. How does one know which should be which?

Reply:
> I was very confused what's the difference between properties and custom properties. Looking at current data tfx pipelines log. Most of executions' properties are properties, while most of artifacts' properties are custom properties. How does one know which should be which?

Previously, both Artifacts and Executions were using properties. Then someone in a TFX team just changed the Artifact properties to custom properties. This broke our UX, but I fixed it. They have not made the same change for Executions, so executions still use properties instead of custom properties.

In MLMD, properties have schema which is defined in the ArtifactType or ExecutionType. Custom properties are more like system properties that are not part of the ArtifactType schema.

Reply:
The long-term plan will be migrating to custom property, so perhaps query both places?

Reply:
I see, I'll create an issue to keep track of these

Reply:
Created https://github.com/kubeflow/pipelines/issues/3493

Reply:
So, shall we keep this as a custom property? I can add reading from custom property in UI additionally.

Reply:
> So, shall we keep this as a custom property? I can add reading from custom property in UI additionally.

If it's not hard for you, then can you please read both property and custom_property in the UI?
It as easy to read custom property or both in UI, but writing property in Metadata Writer has more consequences that writing custom_property (due to schema).

Reply:
sgtm, I will change UI to read from both, you can unhold to merge

Reply:
/hold cancel